src/Expression/Simplifier.purs

```diff
@@ -27,12 +27,6 @@ simplify (ExpressionBinary operation leftExpression rightExpression) =
 simplify expression = expression
 
 trimTimesNodes :: Expression -> Expression -> Maybe Expression
-trimTimesNodes (ExpressionLiteral leftValue) (ExpressionLiteral rightValue) =
-  if (toRational 0) == leftValue || (toRational 0) == rightValue then
-    Just $ ExpressionLiteral (toRational 0)
-  else
-    Nothing
```

Comment:
I think it is safe and beneficial to rewrite a product of two literals with a single literal holding the Rational product of the two values.  I do not feel strongly about this but unless there is a reason not to do it, I recommend to add such rule instead of the rule you deleted, eg the rule specific to multiplication by 0.

Reply:
This should be performed anyway during the `trimConstantLeafNodes` function on line 116